### PR TITLE
HZN-185: Document Installation on Debian-based system

### PR DESCRIPTION
**Pull request by indigo423**
Create installation description to install OpenNMS on a Debian based system.
The setup describes the installation with the opennms meta package which handles
all dependencies like JICMP, PostgreSQL and Oracle Java.

Created generic introduction about the overall installation workflow and how
different releases can be installed.

JIRA: http://issues.opennms.org/browse/HZN-185
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS119/latest

### CODE REVIEW

File: opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                              Branches can be found in http://yum.opennms.org/branches/ or http://debian.opennms.org/dists/branches/
+| `branches/${RELEASE}`     | Install a specific release, e.g. `branches/release-14.0.3`.
+                              This release branches are also found in http://yum.opennms.org/branches/ or http://debian.opennms.org/dists/branches/
+|===
+
+To install a different release the repository files have to be installed and manually modified.
+
+==== Specific Release on RHEL-based system
+
+.RHEL-based system change the repository base URL in the file
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc]: This is not a sentence and what file has to be changed?

File: opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+[source, shell]
+----
+deb http://debian.opennms.eu snapshot main <1>
+deb-src http://debian.opennms.eu snapshot main <1>
+----
+
+<1> Change repository path in this example from `stable` to `snapshot`
+
+Run `apt-get update`and install _OpenNMS_ with _apt_ following the normal installation procedure.
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc]: Missing space at "update`and"

File: opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+[opennms-stable-rhel7]
+name=RedHat Enterprise Linux 7.x and CentOS 7.x (snapshot)<1>
+baseurl=http://yum.opennms.org/snapshot/rhel7<2>
+failovermethod=roundrobin
+gpgcheck=1
+gpgkey=file:///etc/yum.repos.d/OPENNMS-GPG-KEY
+----
+
+<1> Change description in this example from `stable` to `snapshot`
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc]: Isn't it in the past so "Changed"

File: opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+<2> Change the _URL_ to use _RPM_ packages in subdirectory `snapshot` instead of `stable`
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc]: Same here

File: opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+vi /etc/apt/sources.list.d/opennms.list
+----
+
+[source, shell]
+----
+deb http://debian.opennms.eu snapshot main <1>
+deb-src http://debian.opennms.eu snapshot main <1>
+----
+
+<1> Change repository path in this example from `stable` to `snapshot`
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/introduction.adoc]: Also here

File: opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[gi-install-opennms-debian]]
+=== Install on Debian-based systems
+
+This section describes how to install the _OpenNMS_ platform on _Ubuntu 14.04 LTS_.
+The setup process is described in the following steps:
+
+. Install OpenNMS apt repository server with GPG key to verify packages
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc]: OpenNMS must be _OpenNMS_

File: opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+. Installation of the _opennms_ meta package which handles all dependencies
+. Initialize _PostgreSQL_ database and configure access
+. Initialize _OpenNMS_ and first start of the application
+
+[[gi-install-opennms-deb-repo]]
+==== Setup OpenNMS Debian repository
+
+.Content of the new _OpenNMS Debian_ repository file `/etc/apt/sources.list.d/opennms.list`
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc]: Better: "Create a new file `/etc/apt/sources.list.d/opennms.list`for the _OpenNMS Debian_ repository:"

File: opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+.Update apt repository cache
+[source, shell]
+----
+apt-get update
+----
+
+[[gi-install-opennms-deb-package]]
+==== Install OpenNMS package
+
+.Installation of the full application including all dependencies like _PostgreSQL_ or _Java_
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc]: "or Java" -> "and Java:"

File: opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+                    user-name="** YOUR-OPENNMS-USERNAME **"
+                    password="** YOUR-OPENNMS-PASSWORD **" />
+
+<jdbc-data-source name="opennms-admin"
+                    database-name="template1"
+                    class-name="org.postgresql.Driver"
+                    url="jdbc:postgresql://localhost:5432/template1"
+                    user-name="** YOUR-OPENNMS-USERNAME **"
+                    password="** YOUR-OPENNMS-PASSWORD **" />
+----
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc]: Maybe also use callouts <?> here, too

File: opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+[[gi-install-opennms-deb-init]]
+==== Initialize OpenNMS
+
+_OpenNMS_ is now configured to access the database.
+It is required to set the _Java_ environment running _OpenNMS_ and initialize the database schema.
+
+.Set _Java_ environment
+[source, shell]
+----
+${OPENNMS_HOME}/bin/runjava -s
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc]: Why not use "/usr/share/opennms" here? It's the debian-specific manual.

File: opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+----
+
+.Initialize database and detect path to libraries
+[source, shell]
+----
+${OPENNMS_HOME}/bin/install -dis
```
Comment (christianpape) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc]: Same here

File: opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc (hunk 10 above)
Comment (indigo423) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc]: Just want to get people used to ${OPENNMS_HOME} so we don't have to differentiate all the time in several other places.

File: opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc (hunk 11 above)
Comment (indigo423) [inline opennms-doc/guide-install/src/asciidoc/text/opennms/debian.adoc]: Same here :)

**Pull request merged**